### PR TITLE
[SYSTEMML-1923] Support sinh, cosh and tanh as built-in functions

### PR DESCRIPTION
**Pull request by niketanpansare**
- Added sinh, cosh and tanh builtin functions to CP, Spark and GPU backend.
- Added these functions to Python DSL.
- Also, updated the DML language reference and Python reference documentation.

@bertholdreinwald @nakul02 can you please review this PR ?

@dusenberrymw We can extend `nn/layers/tanh.dml` to use this function.

### CODE REVIEW

File: src/main/cpp/kernels/SystemML.cu [hunk 1]
```diff
@@ -1018,6 +1046,20 @@ __global__ void matrix_tan(double *A, double *C, unsigned int size) {
...
+ * Do an tanh over all the elements of a matrix
+ * @param A the input matrix (of length = size)
+ * @param C the pre-allocated output matrix (of length = size)
+ * @param siz the length of the input and output matrices
+ */
+extern "C"
+__global__ void matrix_tanh(double *A, double *C, unsigned int size) {
+    int index = blockIdx.x * blockDim.x + threadIdx.x;
+    if (index < size){
+        C[index] = tan(A[index]);
```
Comment (nakul02) [inline src/main/cpp/kernels/SystemML.cu]: `tanh`?

File: docs/dml-language-reference.md [hunk 2]
```diff
@@ -822,7 +822,7 @@ is same as
 Function | Description | Parameters | Example
 -------- | ----------- | ---------- | -------
 exp(), log(), abs(), sqrt(), round(), floor(), ceil() | Apply mathematical function on input (cell wise if input is matrix) | Input: (&lt;matrix&gt;), or (&lt;scalar&gt;) <br/> Output: &lt;matrix&gt;, or &lt;scalar&gt; | sqrt(X) <br/> log(X,y) <br/> round(X) <br/> floor(X) <br/> ceil(X)
-sin(), cos(), tan(), asin(), acos(), atan() | Apply trigonometric function on input (cell wise if input is matrix) | Input: (&lt;matrix&gt;), or (&lt;scalar&gt;) <br/> Output: &lt;matrix&gt;, or &lt;scalar&gt; | sin(X)
+sin(), cos(), tan(), sinh(), cosh(), tanh(), asin(), acos(), atan() | Apply trigonometric function on input (cell wise if input is matrix) | Input: (&lt;matrix&gt;), or (&lt;scalar&gt;) <br/> Output: &lt;matrix&gt;, or &lt;scalar&gt; | sin(X)
```
Comment (nakul02) [inline docs/dml-language-reference.md]: Thank you for updating the doc

File: src/main/cpp/kernels/SystemML.cu (hunk 1 above)
Comment (niketanpansare) [reply src/main/cpp/kernels/SystemML.cu]: Thanks for catching this :+1:

**Comment by niketanpansare** [comment]
`org.apache.sysml.test.gpu.UnaryOpTests` are successful. Also, tested the results using simple DML script with `-exec spark` and compared it with `-gpu`.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2112/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2112/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2112/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.indexing/RightIndexingMatrixTest/testRightIndexingSparseSP/'><strong>org.apache.sysml.test.integration.functions.indexing.RightIndexingMatrixTest.testRightIndexingSparseSP</strong></a></li></ul>

**Comment by mboehm7** [comment]
could you please put a TODO into `codegen.cplan.CNodeUnary` - I'll take care of the rest.

**Comment by nakul02** [comment]
GPU related changes LGTM :+1:

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2117/

**Comment by dusenberrymw** [comment]
@niketanpansare good to know, thanks!

**Pull request closed without merging**